File: app/src/main/java/com/llbt/meepwn/lincolnblock/main/view/adapter/RecycleViewAdapter.java
package com.llbt.meepwn.lincolnblock.main.view.adapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.llbt.meepwn.lincolnblock.R;
import com.llbt.meepwn.lincolnblock.databinding.ItemBinding;
import com.llbt.meepwn.lincolnblock.main.view_model.TestItemViewModel;

import java.util.List;

/**
 * package: com.llbt.meepwn.lincolnblock.main.view.adapter
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/29 下午4:20
 * desc:
 */
public class RecycleViewAdapter extends RecyclerView.Adapter<RecycleViewAdapter.ViewHolder> {

	private List<TestItemViewModel> dataList;
	private Context context;
	private LayoutInflater inflater;

	public RecycleViewAdapter(Context context, List<TestItemViewModel> dataList) {
		this.dataList = dataList;
		this.context = context;
		this.inflater = LayoutInflater.from(context);
	}

	@Override
	public RecycleViewAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		ItemBinding binding = DataBindingUtil.inflate(inflater, R.layout.item, parent, false);
		RecycleViewAdapter.ViewHolder holder = new ViewHolder(binding.getRoot());
		holder.binding = binding;
		return holder;
	}

	@Override
	public void onBindViewHolder(RecycleViewAdapter.ViewHolder holder, int position) {
		TestItemViewModel viewModel = dataList.get(position);
		holder.binding.setViewModel(viewModel);
		holder.binding.executePendingBindings();
	}

	@Override
	public int getItemCount() {
		return dataList.size();
	}

	class ViewHolder extends RecyclerView.ViewHolder {
		ItemBinding binding;

		public ViewHolder(View itemView) {
			super(itemView);
		}
	}

}

File: app/src/main/java/com/llbt/meepwn/lincoinblock/main/view/adapter/RecycleViewAdapter.kt
package com.llbt.meepwn.lincoinblock.main.view.adapter

import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.llbt.meepwn.lincoinblock.R
import com.llbt.meepwn.lincoinblock.databinding.ItemBinding
import com.llbt.meepwn.lincoinblock.main.view_model.ItemViewModel

/**
 * package: com.llbt.meepwn.lincoinblock
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/12 上午11:50
 * desc:
 */
class RecycleViewAdapter(private val context: Context, private val list: List<ItemViewModel>) : RecyclerView.Adapter<RecycleViewAdapter.ViewHolder>() {
    private val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = DataBindingUtil.inflate<ItemBinding>(inflater, R.layout.item, parent, false)
        val holder = ViewHolder(binding.root)
        holder.binding = binding
        return holder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val viewModel = list[position]
        holder.binding?.viewModel = viewModel
        holder.binding?.executePendingBindings()
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var binding: ItemBinding? = null
    }

}
